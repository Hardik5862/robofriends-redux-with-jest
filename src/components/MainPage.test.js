import React from "react";
import { shallow } from "enzyme";
import MainPage from "./MainPage";

let wrapper;
beforeEach(() => {
  const mockProps = {
    onRequestRobots: jest.fn(),
    robots: [],
    searchField: "",
    isPending: false,
  };

  wrapper = shallow(<MainPage {...mockProps} />);
});

it("renders MainPage correctly", () => {
  expect(wrapper).toMatchSnapshot();
});

it("filter robots correctly when array is empty", () => {
  expect(wrapper.instance().filteredRobots()).toEqual([]);
});

it("filter robots correctly with search term a", () => {
  const mockProps2 = {
    onRequestRobots: jest.fn(),
    robots: [
      {
        id: 1,
        name: "Vision",
        email: "vision@gmail.com",
      },
    ],
    searchField: "a",
    isPending: false,
  };

  const wrapper2 = shallow(<MainPage {...mockProps2} />);
  expect(wrapper2.instance().filteredRobots()).toEqual([]);
});

it("filter robots correctly with search term v", () => {
  const mockProps2 = {
    onRequestRobots: jest.fn(),
    robots: [
      {
        id: 1,
        name: "Vision",
        email: "vision@gmail.com",
      },
    ],
    searchField: "v",
    isPending: false,
  };

  const wrapper2 = shallow(<MainPage {...mockProps2} />);
  expect(wrapper2.instance().filteredRobots()).toEqual([
    {
      id: 1,
      name: "Vision",
      email: "vision@gmail.com",
    },
  ]);
});
