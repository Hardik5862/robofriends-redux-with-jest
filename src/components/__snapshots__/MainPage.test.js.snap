// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MainPage correctly 1`] = `
<div
  className="tc"
>
  <Header />
  <SearchBox />
  <Scroll>
    <CardList
      robots={Array []}
    />
  </Scroll>
</div>
`;
